account_move_journal_template/models/account_move_journal_template.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2022 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
+from odoo.tools import UserError
+
+
+class AccountMoveJournalTemplate(models.Model):
+    _name = "account.move.journal.template"
+    _description = "Template for additional journal entries/items"
+
+    _sql_constraints = [
+        (
+            "product_tmpl_id",
+            "UNIQUE (product_tmpl_id)",
+            _("Template for this product already exists"),
+        ),
+    ]
+
+    name = fields.Char(required=True, copy=False, default="New")
+    product_tmpl_id = fields.Many2one(
+        comodel_name="product.template",
+        string="Product",
+        help="Journal items will be created for this product",
+    )
+    journal_item_ids = fields.One2many(
+        comodel_name="account.move.journal.template.item.line",
+        inverse_name="journal_template_id",
+        copy=True,
+        string="Journal Items",
+        help="Journal items to be added in new journal entry",
+    )
+    active = fields.Boolean(default=True)
+
+    def action_toggle_active(self):
+        self.ensure_one()
+        self.active = not self.active
+
+    def _check_balanced(self):
+        """ An adaptation of account.move._check_balanced()"""
+        moves = self.filtered(lambda move: move.journal_item_ids)
+        if not moves:
+            return
+        self.env["account.move.journal.template.item.line"].flush(
+            self.env["account.move.journal.template.item.line"]._fields
+        )
+        self._cr.execute(
```

Comment:
The currency on the journal can override the currency of the company. Maybe best to setup a for this in self iteration and test template for template. Then determine currency from journal and company. Inject the currency.decimal_places as a variable in the SQL statement. On the plus side this will safe some joins. All this rounding and manipulation would not be needed if Odoo used proper decimal fields instead of floats to store monetary amounts.

Reply:
I will need to revisit this, because it is not yet completely clear to me what to do